### ISSUE TITLE
windows support?

### ISSUE THREAD

**Issue by ajkessel** [gchamon/sysrsync#43]
Has anyone tried to get this working on Windows? You obviously need a system rsync but that is available via cygwin (or packaged separately [here](https://www.itefix.net/cwrsync/client/downloads). Is there a more straightforward to do this?

**Comment by gchamon** [comment]
I don't see why it shouldn't work on WSL2, since it's a fully fledged Linux working in a VM, but I don't have the OS neither the resources to test this

**Comment by ajkessel** [comment]
Yes, it's fine on WSL. But for a Windows-Python end user it won't work out of the box since there is no Windows-native rsync, and if you manually install Cygwin rsync, there are problems with filesystem mapping and finding ssh and keyfile. Wondering if anyone has a more elegant solution to make this module fully cross-platform.

**Comment by gchamon** [comment]
I'm going to do a quick research sometime to see what are the challenges of supporting windows natively. Priority for me right now is to get the ball rolling for this repo again, which means taking a look at interface, abstractions and tests. Get the fundamentals right.

**Comment by ajkessel** [comment]
There might be a way to leverage/package Cygwin rsync. I ended up coding a very poor-man's rsync in Python for purposes of Windows compatibility but would be happy to help/test a real implementation.

**Comment by gchamon** [comment]
Guess I just have to come up with the tests

![image](https://github.com/user-attachments/assets/ab5f3cf4-1642-4764-a412-da88cfd9db2c)
